src/Integration/State/StateManager.cs

```diff
@@ -109,14 +108,7 @@ public IEnumerable<ConnectionInformation> GetConnectedServers()
 
         public void SetProjects(ConnectionInformation connection, IEnumerable<SonarQubeProject> projects)
         {
-            if (this.Host.UIDispatcher.CheckAccess())
-            {
-                this.SetProjectsUIThread(connection, projects);
-            }
-            else
-            {
-                this.Host.UIDispatcher.BeginInvoke(new Action(() => this.SetProjectsUIThread(connection, projects)));
-            }
```

Comment:
This is the only material product change.

There weren't any tests that checked that the Dispatcher was used.
I looked at adding one to check that the thread handler is called but the `StateManager` is old code that wasn't written with testing in mind, and I couldn't see a straightforward way to do it.

Instead, I've added a thread check in line 167 below. Also, I've manually debugged through the code to check what happens when setting up a new binding. This method is called on the UI thread and from a background thread during binding, and from a UI thread when a solution is closed. In all cases the `SetProjectsUIThread` is called on the UI thread.